### ISSUE TITLE
[Track 4] 모델용량 기준 문의

### ISSUE THREAD

**Issue by 2KangHo** [AI-GrandChallenge/round-1#15]
https://github.com/AI-GrandChallenge/round-1/blob/f3c9dff78e1d642c052bafa3731e3bcb217517b7/iitp_category/main.py#L91
https://github.com/AI-GrandChallenge/round-1/blob/f3c9dff78e1d642c052bafa3731e3bcb217517b7/iitp_category/main.py#L92

위 코드를 보면 모델 용량이 tensor 개수로 정해질텐데 이러면 처음 모델 경량화, 최적화의 주최의도와 많이 달라지지 않나 생각이 드는데 명확한 기준 부탁드립니다.

**Comment by chs1733** [comment]
IITP의 최종안내서에 보면,
모델 용량 기준이 weights 파일 용량이라고 정의되어 있는데,
저건 parameter숫자이지 않습니까?
parameter숫자와 weights 파일 용량은 전혀 다른 개념입니다. 사이즈가 10배 이상 차이가 납니다.

최종안내서가 잘못된거라면 지금이라도 재공지를 해야 혼란을 막을 수 있지 않을까요?
최종 안내서를 기준으로 weights 파일 용량 기준으로 개발했는데, 또 다시 개발해야 하는건가요?
평가의 명확한 기준을 정의해주시기 바랍니다.

**Comment by 2KangHo** [comment]
@chs1733 저희도 용량기준으로 했는데 최종안내서에도 뭔가 run시에 압축 해제하는 건 안된다고 하여 다른 weight압축기법을 사용할 수 없을 것 같아서 답변만 기다리고 있는데 금요일 밤부터 아무런 응답이 없네요

**Comment by bluebrush** [comment]
혼란을 드렸다니 죄송합니다. 
정리하면 아래와 같습니다. 
1. 상품데이터 분류 문제로 푼다.
2. 제출하는 모델은 15MB(저장 크기)보다 작아야한다
3. 4byte float 기준으로 3.75M 파라미터보다 작게하거나, 8bit 퀀타이즈 하거나 해서 15MB 이하로 저장하면 됩니다. 
4. resnet18의 모델을 퀀타이즈해도 되고, 다른 구조의 모델을 사용해도 됩니다.

**Comment by JeiKeiLim** [comment]
> 혼란을 드렸다니 죄송합니다.
> 정리하면 아래와 같습니다.
> 
> 1. 상품데이터 분류 문제로 푼다.
> 2. 제출하는 모델은 15MB(저장 크기)보다 작아야한다
> 3. 4byte float 기준으로 3.75M 파라미터보다 작게하거나, 8bit 퀀타이즈 하거나 해서 15MB 이하로 저장하면 됩니다.
> 4. resnet18의 모델을 퀀타이즈해도 되고, 다른 구조의 모델을 사용해도 됩니다.

파라미터가 3.75M 넘어도 15메가 아래로 내려갈 수 있지 않나요?
기준이 확실했으면 좋겠습니다.
기준을 2가지 사용하는건가요?

**Comment by ClovaAIAdmin** [comment]
@JeiKeiLim 
산술적으로 그렇다고 보시면 됩니다.. 파라미터 수로만 제한하면 quantization 을 할 이유가 없기도 하고
실제 디바이스 포팅시에는 파일크기가 중요한 이슈이기 때문이며
기준은 단 하나 "압축-해제를 하지 않고 checkpoint 파일 기준 15MB 이하로 파일크기를 가진 모델을 기준한다." 입니다.

**Comment by 2KangHo** [comment]
@ClovaAIAdmin 그렇다면 huffman code나 CSR등의 weight 저장도 불허하는 건가요? 그리고 npz_compressed와 같은 함수를 통해 pruning의 효과를 극대화 하는 경우도 불허되는 건가요?

**Comment by JeiKeiLim** [comment]
> @JeiKeiLim
> 산술적으로 그렇다고 보시면 됩니다.. 파라미터 수로만 제한하면 quantization 을 할 이유가 없기도 하고
> 실제 디바이스 포팅시에는 파일크기가 중요한 이슈이기 때문이며
> 기준은 단 하나 "압축-해제를 하지 않고 checkpoint 파일 기준 15MB 이하로 파일크기를 가진 모델을 기준한다." 입니다.

사용하는 Framework 에서 load 함수를 이용하여 불러올 수 있는 모델 파일 크기만 인정한다는 이야기로 보면 되나요?
뭘로 정해져도 그에 맞추면 될텐데 확실한 기준이 생겼으면 좋겠습니다.

**Comment by ClovaAIAdmin** [comment]
@JeiKeiLim @2KangHo 
네 맞습니다. 여러가지 방법을 쓰기 시작하면 각 케이스 별로 예외처리를 해야하는 상황이 발생합니다. 
이번 예선에서는 연산량의 감소를 추구하면서도 일괄적이면서 명시적으로 통일할 수 있는 방법인
 "framework의 기본로딩 함수로 불러오는 파일의 크기" 만 인정한다로 보시면 되겠습니다.

**Comment by 2KangHo** [comment]
@ClovaAIAdmin 늦은 시간 답변 감사합니다. 그렇다면 pytorch의 경우 torch.load로 불러들일 수 있는 형태의 파일은 다 허용한다는 뜻으로 이해하면 될까요?

**Comment by ClovaAIAdmin** [comment]
@2KangHo 
네 다만.. 말씀드린 바와 같이 압축 파일이면 안됩니다.

**Comment by 2KangHo** [comment]
@ClovaAIAdmin 네 감사합니다. 그럼 간단하게 bind_model(model) 함수를 변경하지 않는 선에서 해결하라는 의미로 이해하면 될까요??

**Comment by ClovaAIAdmin** [comment]
@2KangHo 
네 그렇습니다~

**Comment by 2KangHo** [comment]
@ClovaAIAdmin 답변 감사합니다! 다른 사람들도 볼 수 있게 close는 내일이나 모레까지는 안하도록 하겠습니다.

**Comment by kjy12345** [comment]
@ClovaAIAdmin "framework의 기본로딩 함수"를 쓰면 된다고 하셨는데, 기본로딩 함수라는 것도 좀 애매한 부분이 있어서요.
먼저 torch의 경우 torch.jit.load도 사용 가능할까요?
또한 tensorflow (또는 tf.keras)의 경우엔, TensorFlow Lite로 convert 하여 inference 하는게 가능한데, 이건 허용될까요?

**Comment by ClovaAIAdmin** [comment]
@kjy12345 
안녕하세요. 위에도 언급했듯이 압축 기능이 포함되지 않은 기본 제공 로더로 이해주시면 되겠습니다. 그런 관점에서 TensorFlow Lite로 convert는 기본적으로 압축옵션을 제공하므로 적절하지 않겠지요. 말씀 주신 대로 공지 형태로 다시 이슈만들겠습니다. 의견 감사합니다.
